### ISSUE TITLE
Keep phone screen from locking?

### ISSUE THREAD

**Issue by rodrigograca31** [steveseguin/vdo.ninja#1002]
Hi!

if vdo.ninja is being used on a phone (latest chrome and latest Android) as audio only mode (no video) the phone screen will lock after X amount of seconds (as normal)

because of that Chrome/Android will stop capturing audio a few seconds later. at least for me on MIUI.

This is annoying as I might want to not use video continue to use the audio....
I dont think we can solve the audio cutting part as its a security feature from Android (to not allow sites to monitor your audio in the background) but we could keep it from locking allowing us to use the phone as a mic.

[I've found this code](https://github.com/richtr/NoSleep.js) which seems to basically create a video element and play it so that the screen doesnt lock. thats what already happens when we are using the camera but im asking for when its audio only. :)

**Comment by steveseguin** [comment]
Thank you for the issue report and suggestion.

I'll look into adding this as an option.

To be clear, this is on the publisher side and not the viewer side?

If looking to capture audio-only, but still have a video active on the publisher side, ?view=xxxx&noaudio might work for your needs in the short term.

**Comment by rodrigograca31** [comment]
Yes its on the publisher side.

Use case: you want to use your phone as a microphone only. so you disable video... but on Android after X seconds it will lock your screen and after a bit longer disable the mic (for security purposes) and I want it not to do that. to stay on and work as a mic

**Comment by steveseguin** [comment]
If you want to try out my first attempt at this, please load up https://vdo.ninja/alpha/  (the alpha version) to see if it does what you want.

Seems to work for my Android phone here, but more testing is probably needed. Thank you for the request.

**Comment by rodrigograca31** [comment]
Yes! Got it going for like 15 mins without turning the screen off! :partying_face: 

Mind showing me the code? I couldnt find the commit for this, im just curious :)

**Comment by steveseguin** [comment]
Glad to hear it.

This is the code.  I just insert a tiny video into the page that the user can't see, that loops, simulating video playing.   The video file itself can just be a small transparent webm file, but for simplicity reasons I just made a tiny black mp4 file.
```
// if a mobile phone, after a user gesture, then
		let fakeElement = document.createElement("video");
		fakeElement.autoplay = true;
		fakeElement.loop = true;
		fakeElement.muted = true;
		fakeElement.src = "./media/micro.mp4"; // 20-kbps 3-second blank/silent video
		fakeElement.style.width = "1px";
		fakeElement.style.height ="1px";
		fakeElement.controls = false;
        document.body.appendChild(fakeElement);
        ```

**Comment by rodrigograca31** [comment]
Nice! :partying_face: 
Feel free to close this issue when its implemented in the main version :)

**Comment by rodrigograca31** [comment]
Hey, this feature has been working great but i noticed a few things:

after using the site (ending the call) it keeps saying "a site is using your mic" (Chrome on Android)

also i learned theres an api called "wakelock" for this cases. but support is "meh"  70%
https://caniuse.com/?search=wakelock
theres also a lib to help: https://www.npmjs.com/package/nosleep.js

example at the bottom: https://rapidapi.com/guides/learn-screen-wake-lock-api
example 2: https://richtr.github.io/NoSleep.js/example/

more info: https://mikevdv.dev/blog/2022-06-23-stop-the-screen-going-to-sleep-with-javascript
https://www.educative.io/answers/how-to-keep-your-screen-awake-using-javascript

**Comment by AlpineSkate** [comment]
I also want to use a phone as my microphone and I've got/had this problem, but I've made the following observation (mobile phones Xiaomi and Samsung):
Disable video, enable audio and then "Start" the screen will lock after the display sleep setting is reached.
Enable video, enable audio, "Start" and then just click on the disable video icon once everything is running and one gets the desired result.
Seeing as I've just discovered this I'm now checking to see if there are negatives to this solution.

**Comment by rajaramaniyer** [comment]
Today morning I faced this issue on my Android phone. I was streaming audio from my Samsung Galaxy S23+ via Chrome.. when screen turns off the streaming will stop.. I had to keep touching the screen in regular intervals until the session was over .. 

Please educate me on what should I do ..

**Comment by steveseguin** [comment]
If you turn the screen off, Chrome will stop the streaming.

While VDO.Ninja will prevent the browser from going to sleep in most cases, if it does go to sleep, it stops.

You can get around this by using the native VDO.Ninja app instead, which tries to support this functionality. However, it won't always work and it's fairly limited versus the web version.

https://docs.vdo.ninja/steves-helper-apps/native-mobile-app-versions

**Comment by rajaramaniyer** [comment]
thanks for the quick response. this is useful. I was looking for something like this, like electroncapture for andriod ..

**Comment by rajaramaniyer** [comment]
But even in the native mobile app, the stream stops when I turn off the screen ..

**Comment by steveseguin** [comment]
https://play.google.com/store/apps/details?id=flutter.vdo.ninja

For Android 14+ specifically, they may need to check:

Settings > Apps > VDO.NInja > Battery > Unrestricted
Settings > Battery > Background battery usage > VDO.NInja > Unrestricted

There may be other settings you can check, however I've just tested it on Android 13 Pixel 4a and it stay active for over a minute, even with the device locked.

I do have a second version of the native app, written from the ground up in a different language, so you can try that too -- it's the one that has the HDMI support.  Give it a go as well
